### PR TITLE
Matchmaker role

### PR DESCRIPTION
**Pull request by imadtg**
hopefully this is the last PR, have done some testing and it works just fine

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -385,6 +385,8 @@ def check(m):
         if len(session[1]) == 0:
             await client.change_presence(game=client.get_server(WEREWOLF_SERVER).me.game, status=discord.Status.online)
     await send_lobby(leave_msg)
+    if member in list(session[1]) and session[0]:
```
Comment (belguawhale) [inline bot.py]: What's the purpose of this? Member isn't even defined outside of the for loops and each member would have already been removed from the game on line 382

File: bot.py [hunk 2]
```diff
@@ -847,6 +851,51 @@ def check(m):
...
+    if not session[0] or get_role(message.author.id, 'role') not in COMMANDS_FOR_ROLE['choose'] or not session[1][message.author.id][0] or not message.channel.is_private:
+        return
+    if parameters == "":
+        await reply(message, roles[session[1][message.author.id][1]][2])
+    else:
+        if get_role(message.author.id, 'role') == 'matchmaker' and 'match' in session[1][message.author.id][4] and not session[2]:
+            parameters = parameters.lower().split(" and ")
+            if len(parameters) == 2:
+                player1 = get_player(parameters[0])
+                player2 = get_player(parameters[1])
```
Comment (belguawhale) [inline bot.py]: Add an error message if two players were not chosen? (currently it just returns if the command `match 1` is used)

File: bot.py [hunk 3]
```diff
@@ -2132,7 +2181,8 @@ def win_condition():
         return None
     
     for player in session[1]:
-        if get_role(player, 'actualteam') == win_team:
+        lover = o.split(':')[1]
```
Comment (belguawhale) [inline bot.py]: What's o? Doesn't seem to be defined above

File: bot.py [hunk 4]
```diff
@@ -2935,6 +3015,14 @@ def sort_roles(role_list):
                     await send_lobby(random.choice(lang['hastotems']).format('**, **'.join([get_name(x) for x in totem_holders[:-1]]), get_name(totem_holders[-1])))
 
             for player in killed_temp:
+                lover = o.split(':')[1]
+                if lover in killed_temp and player in killed_temp:
+                    #fix for lover suicide message appears if player dies even tho lover died already
+                    session[1][player][4].remove('lover:' + lover)
+                    session[1][lover][4].remove('lover:' + player)
+                    await player_death(player, 'night kill')
```
Comment (belguawhale) [inline bot.py]: uhh doesn't this make night kills not cause the other lover to die?

File: bot.py [hunk 5]
```diff
@@ -3132,7 +3220,8 @@ def sort_roles(role_list):
                      'shoot' : ['gunner'],
                      'observe' : ['werecrow', 'sorcerer'],
                      'pass' : ['harlot', 'hunter'],
-                     'id' : ['detective']}
+                     'id' : ['detective']
```
Comment (belguawhale) [inline bot.py]: I think you're missing a comma here

File: bot.py [hunk 6]
```diff
@@ -3212,7 +3305,9 @@ def sort_roles(role_list):
             'cursed villager' :
             [0, 0, 1, 1, 1, 1,  1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 3],
             'gunner' :
-            [0, 0, 0, 0, 0, 0,  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+            [0, 0, 0, 0, 0, 0,  1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+            'matchmaker' :
+            [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
```
Comment (belguawhale) [inline bot.py]: Did you `!verifygamemodes` to make sure role counts are correct?

File: bot.py [hunk 7]
```diff
@@ -667,6 +669,8 @@ def check(m):
                                    "`{}role gunner` for more information.".format(
                             session[1][player][4].count('bullet'), '' if session[1][player][4].count('bullet') == 1 else 's',
                             BOT_PREFIX))
+                    if role == 'matchmaker' and sendrole:
```
Comment (belguawhale) [inline bot.py:709]: You can add matchmaker to the list of roles on line 664 and this will automatically be factored in

Comment (imadtg) [reply bot.py:709]: but i want it to only be dmed once

File: bot.py (hunk 3 above)
Comment (imadtg) [reply bot.py]: oh wait

File: bot.py (hunk 4 above)
Comment (imadtg) [reply bot.py]: this only happens if both lovers were killed at night, and that is a fix for this 
tempae

File: bot.py (hunk 7 above)
Comment (belguawhale) [reply bot.py:709]: Oh, right

File: bot.py (hunk 2 above)
Comment (imadtg) [reply bot.py]: done

File: bot.py (hunk 3 above)
Comment (imadtg) [reply bot.py]: fixed

File: bot.py (hunk 6 above)
Comment (imadtg) [reply bot.py]: fixed too

File: bot.py (hunk 7 above)
Comment (belguawhale) [reply bot.py:709]: Personally i'd still like to be dmed the list of living players even after my ability is used, like hunter but that's my personal opinion

File: bot.py [hunk 8]
```diff
@@ -3027,7 +3141,15 @@ def sort_roles(role_list):
...
-                        await end_game(win_msg, [lynched_player])
+                        o = []
+                        for n in session[1][lynched_player][4]:
+                            if n.startswith('lover:'):
+                                o.append(n.split(':')[1])
+                        if o:
+                            lovers = o
+                        else:
+                            lovers = []
+                        await end_game(win_msg, [lynched_player] + lovers)
```
Comment (belguawhale) [inline bot.py:3271]: If matched fool is lynched, their lover doesn't win except in random gamemode. Just for the future

Comment (imadtg) [reply bot.py:3271]: it isn't like fool and matchmaker have mutual gamemodes other then random for now \:P

**Review by belguawhale — APPROVED** [review]

**Comment by belguawhale** [comment]
Matchmaker role! (thanks to imadtg)

**Pull request merged**